packages/data-service/src/data-service.ts

```diff
@@ -1996,6 +2056,63 @@ export class DataServiceImpl extends EventEmitter implements DataService {
     return !!this._metadataClient;
   }
 
+  isOperationCancelledError(error: Error): boolean {
+    return OperationCancelledError.isOperationCancelledError(error);
+  }
+
+  private async cancellableOperation<T>(
```

Comment:
I can’t really find anything on *why* we are doing this in the tech design/ticket/PR description. What makes this better than having the cancellation feature implemented outside of data-service? Are there follow-up tickets to migrate the other cancelable-query features to this API as well?

My understanding is that it can be implemented on top of the data-service API, and so even if we have the desire for a shared implementation, it wouldn’t be something that needs to be part of the `DataService` interface itself and could be turned into a separate helper (which would also have the advantage that other providers of the `DataService` interface wouldn’t need to copy this code here).